### ISSUE TITLE
 Setup failed 0x80070643 on Windows server 2016 too

### ISSUE THREAD

**Issue by raa-hasel** [Microsoft/IIS.Administration#287]
we have the same problems like #274 but on W2016. Is there some more or new informations, 
because we need the IIS-Administration Tool for WindowsAdminCenter.
Enclosed our Setup Log
[Microsoft_IIS_Administration_2.3.0_20190922161635_000_iis_administration_msi.log](https://github.com/microsoft/IIS.Administration/files/3639891/Microsoft_IIS_Administration_2.3.0_20190922161635_000_iis_administration_msi.log)

**Comment by bariscaglar** [comment]
@shhsu

**Comment by shhsu** [comment]
After some research, it seems this issue is yet a different issue.

We call chcp 437 to switch the console to US region to workaround Wix localization issues and it seems that the error occurred at this call

@raa-hasel , can you verify that buy running `chcp 437` locally and see if it resuts to an error?

If so, there are some resources online pointing to the same error: [calling chcp resulted in error code 0x80070002](https://social.technet.microsoft.com/Forums/en-US/ec259d7d-07e1-4313-8a2d-3e3d8e980d20/scripterror-bug-in-dns-management-pack-version-10060?forum=operationsmanagermgmtpacks)

There's more inspection needed on client's part to determine what was the exact issue but this is a windows issue. Please let me know your findings

**Comment by shhsu** [comment]
Other customers reported similar issues on chcp
https://www.google.com/search?q=chcp+0x80070002+site:social.technet.microsoft.com&client=firefox-b-1-d&sxsrf=ACYBGNTuU6K41Zdf5BLU0Fm2Ih5lsg7vLA:1569272654960&sa=X&ved=2ahUKEwi-952Z7OfkAhXSv54KHQDRB7MQrQIoAjAAegQIAxAL&biw=2560&bih=1326

**Comment by raa-hasel** [comment]
Hi Peter,
thanks for your reply. I do not know if I start the setup in the right way with chcp 437, but I got the same error.
![grafik](https://user-images.githubusercontent.com/55662020/65509953-fb793080-ded3-11e9-9f92-05368beeab59.png)

Generally our OS is W2016 Std with GUI and German-Languages. What more information do you need of our OS/Installation ? 
And thanks for the links, it seems this is an old windows-issue which is not fixed till today. Otherwise MS has the demand for WAC and IIS control to use the IISAdministraionSetup.exe. 
At least, why do you think that could be a chcp-issue, because the error-log shows:

MSI (s) (2C:58) [13:59:27:341]: Executing op: ActionStart(Name=**CA_PSINSTALL,,)
MSI (s) (2C:58) [13:59:27:341]: Executing op: CustomActionSchedule(Action=CA_PSINSTALL,ActionType=1025,Source=BinaryData,Target=CAQuietExec64,CustomActionData="PowerShell.exe" -NonInteractive -NoLogo -InputFormat None -NoProfile -ExecutionPolicy Bypass -Command "& 'C:\Program Files\IIS Administration\2.3.0\setup\msi-setup.ps1' Install -Verbose -Path 'C:\Program Files\IIS Administration\' -Version '2.3.0'" -IncludeDefaultCors:$true)
MSI (s) (2C:D8) [13:59:27:341]: Invoking remote custom action. DLL: C:\Windows\Installer\MSI229D.tmp, Entrypoint: CAQuietExec64
MSI (s) (2C:30) [13:59:27:341]: Generating random cookie.
MSI (s) (2C:30) [13:59:27:356]: Created Custom Action Server with PID 6096 (0x17D0).
MSI (s) (2C:C8) [13:59:27:387]: Running as a service.
MSI (s) (2C:C8) [13:59:27:387]: **Hello, I'm your 32bit Impersonated** custom action server.
CAQuietExec64:  Error 0x80070002: Command failed to execute.
CAQuietExec64:  Error 0x80070002: QuietExec64 Failed
CAQuietExec64:  Error 0x80070002: Failed in ExecCommon64 method
CustomAction CA_PSINSTALL returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)

May be there are missing 32bit software for the setup in our Installation ?

**Comment by shhsu** [comment]
The reason I think `chcp` failed is because I have a successful installation like this
```
MSI (s) (68:7C) [13:55:58:390]: Generating random cookie.
MSI (s) (68:7C) [13:55:58:593]: Created Custom Action Server with PID 6408 (0x1908).
MSI (s) (68:30) [13:55:58:640]: Running as a service.
MSI (s) (68:78) [13:55:58:640]: Hello, I'm your 32bit Impersonated custom action server.
CAQuietExec64:  Active code page: 437
CAQuietExec64:  VERBOSE: Verifying user is an Administrator
CAQuietExec64:  VERBOSE: Ok
CAQuietExec64:  VERBOSE: Verifying user is an Administrator
CAQuietExec64:  VERBOSE: Ok
CAQuietExec64:  Checking installation requirements
CAQuietExec64:  VERBOSE: Ok
CAQuietExec64:  VERBOSE: Verifying .NET Core shared framework installed
CAQuietExec64:  VERBOSE: Ok
CAQuietExec64:  VERBOSE: Checking if port '55539' is available
CAQuietExec64:  VERBOSE: Ok
```

Since chcp is the fist thing setup calls https://github.com/microsoft/IIS.Administration/blob/dev/scripts/setup/msi-setup.ps1#L122 and `chcp` must generate output like `Active code page: 437` I strongly suspect that `chcp` failed and the catch block didn't capture the error

**Comment by shhsu** [comment]
Looking at your install logs again, I saw this entry
```
MSI (s) (64:0C) [16:16:42:626]: Executing op: CustomActionSchedule(Action=CA_PSINSTALL,ActionType=1025,Source=BinaryData,Target=CAQuietExec64,CustomActionData="PowerShell.exe" -NonInteractive -NoLogo -InputFormat None -NoProfile -ExecutionPolicy Bypass -Command "& **'C:\Program Files\IIS Administration\2.3.0\setup\msi-setup.ps1'** Install -Verbose -Path 'C:\Program Files\IIS Administration\' -Version '2.3.0'" -IncludeDefaultCors:$true)
```

**Comment by shhsu** [comment]
After further investigation I am noticing these lines in the log
```
CustomAction CA_PSINSTALL returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
MSI (s) (64:0C) [16:16:42:736]: Note: 1: 2265 2:  3: -2147287035 
MSI (s) (64:0C) [16:16:42:736]: User policy value 'DisableRollback' is 0
MSI (s) (64:0C) [16:16:42:736]: Machine policy value 'DisableRollback' is 0
Action ended 16:16:42: InstallFinalize. Return value 3.
MSI (s) (64:0C) [16:16:42:736]: Note: 1: 2318 2:  
```
Error code 2265 means `Could not commit storage. System error`, -2147287035 means `Access Denied`

A few other possibilities are there now.
1. Can the disk be out of storage space?
2. Is your environment variable `ProgramFiles` correctly pointing to the 64 bits program files directory?

**Comment by raa-hasel** [comment]
>-Can the disk be out of storage space?
more than 40 GB of storages free, should be enough :-) 
>-Is your environment variable ProgramFiles correctly pointing to the 64 bits program files directory?
![svr141web_path](https://user-images.githubusercontent.com/55662020/65609977-f211c600-dfb0-11e9-8800-b17099c45f93.PNG)

Meantime what I have done:
I make a clean Lab-Installation like my webserver, except url-rewrite, php etc. and the IISAdministrationsetup was running sucessfull.
On my live-webserver I've monitor the setup with ProcMon and found only one issue of "access denied" in C:\Windows\Temp although the ICALs on both server are the same. I gave "everybody" full access and restart the setup, no more "access denied" but atleast without success. In the 2nd. ProcMon-Log i could not find any issue more. So I gave slowly up..... 
Always during the point of CA_PSINSTALL we got the same errors.
